mirrord-layer/src/lib.rs

```diff
@@ -174,29 +176,42 @@ where
     // Stores a list of `oneshot`s that communicates with the hook side (send a message from -layer
     // to -agent, and when we receive a message from -agent to -layer).
     getaddrinfo_handler_queue: VecDeque<ResponseChannel<Vec<AddrInfoInternal>>>,
+
+    pub tcp_steal_handler: TcpStealHandler,
```

Comment:
how about unifying `tcp_mirror_handler` & `tcp_steal_handler` to `tcp_handler`? under something like `Box<dyn TcpHandler>` since both implement `TcpHandler` it can be better than initializing both since only one will be in use

Reply:
That's actually something I tried to do but wasn't sure how exactly (not very experienced with `dyn`). The reason I didn't put any more effort into making it like that is very soon it wont be either, but both based on configurations so no reason to bother so hard ;)

Reply:
![](https://media4.giphy.com/media/1ZkMDj88mQ1rO/giphy.gif)